internal/agent-api/client.go

```diff
@@ -241,6 +248,24 @@ func (a *AgentClient) UptimeMillis() time.Duration {
 	return time.Since(a.workloadStartedAt)
 }
 
+func (a *AgentClient) DeployRequest() *DeployRequest {
+	return a.deployRequest
+}
+
+func (a *AgentClient) IsSelected() bool {
+	return a.selected
+}
+
+func (a *AgentClient) MarkSelected() {
```

Comment:
if these properties shouldn't be set anywhere outside that function, then we should move these property sets into that function and not give developers the chance to footgun

Reply:
They are used in another package.

Reply:
Having to comment something by saying that it should never be called except by one other function is a potential smell. Months from now when we don't know why we did any of this, we could get tripped up here.

Reply:
Comment removed.